### PR TITLE
MGMT-2359: Event log spam fix 

### PR DESCRIPTION
**Pull request by RazRegev**
return forbidden error code when host in installing-pending-user-action tries to register after passing the reboot stage.

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -1577,7 +1577,7 @@ func (b *bareMetalInventory) RegisterHost(ctx context.Context, params installer.
 		uerr := errors.Wrap(err, "Failed to register host: error creating host metadata")
```
Comment (avishayt) [inline internal/bminventory/inventory.go:1577]: @RazRegev We assume here that hostApi.RegisterHost fails because the host already exists?

Comment (RazRegev) [reply internal/bminventory/inventory.go:1577]: We are not assuming it here, the check for that is done in transaction.go, here we just converting the error type to installer.NewRegisterHostForbidden() if it receive error code 403. 
By that agent will stop trying to register.

**Pull request merged**